diff --git a/mk/xen-device-model.spec.in b/mk/xen-device-model.spec.in
index eb6b4ed..d279b32 100644
--- a/mk/xen-device-model.spec.in
+++ b/mk/xen-device-model.spec.in
@@ -7,7 +7,7 @@ Group: System/Hypervisor
 Source0: xen-device-model-%{version}.tar.bz2
 Patch0: xen-device-model-development.patch
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-buildroot
-BuildRequires: SDL-devel, zlib-devel, xen-devel, ncurses-devel, pciutils-devel
+BuildRequires: SDL-devel, zlib-devel, xen-devel, xen-dom0-devel, ncurses-devel, pciutils-devel
 
 %description
 This package contains qemu-dm, the Xen device model.
