# HG changeset patch
# Parent d55764563b44c59ab220469ca7c3befe7f38e14c

diff -r d55764563b44 vnc.c
--- a/vnc.c	Wed May 09 16:52:39 2012 +0100
+++ b/vnc.c	Wed May 09 17:55:39 2012 +0100
@@ -88,7 +88,7 @@ typedef void VncSendHextileTile(VncState
                                 void *last_fg,
                                 int *has_bg, int *has_fg);
 
-#define VNC_MAX_WIDTH 2048
+#define VNC_MAX_WIDTH 2560
 #define VNC_MAX_HEIGHT 2048
 #define VNC_DIRTY_WORDS (VNC_MAX_WIDTH / (16 * 32))
 
